<launch>
      <arg name="sensor_echo_names" default="[echo1]"/>
      <arg name="sensor_echo_topics" default="[airsim/echo1/pointcloud]"/>
      <arg name="sensor_echo_frames" default="[base_echo1]"/>

      <arg name="sensor_lidar_names" default="[lidar1]"/>
      <arg name="sensor_lidar_toggle_groundtruth" default="[1]"/>
      <arg name="sensor_lidar_topics" default="[airsim/lidar1/pointcloud]"/>
      <arg name="sensor_lidar_segmentation_topics" default="[airsim/lidar1/segmentation]"/>
      <arg name="sensor_lidar_frames" default="[base_lidar3]"/>

      <arg name="sensor_gpulidar_names" default="[lidar2]"/>
      <arg name="sensor_gpulidar_topics" default="[airsim/lidar2/pointcloud]"/>
      <arg name="sensor_gpulidar_frames" default="[base_lidar2]"/>

      <arg name="sensor_uwb_names" default="[]"/>
      <arg name="sensor_uwb_topics" default="[airsim/uwbX/diagnostics, airsim/uwbX/range, airsim/uwbX/rangeArray]"/>
      <arg name="sensor_uwb_frames" default="[base_uwb1, base_uwb2]"/>

      <arg name="sensor_camera_names" default="[leftcamera, rightcamera]"/>
      <arg name="sensor_camera_toggle_scene_mono" default="[0, 0]"/>
      <arg name="sensor_camera_scene_quality" default="[0, 0]"/>
      <arg name="sensor_camera_toggle_segmentation" default="[1, 1]"/>
      <arg name="sensor_camera_toggle_depth" default="[1, 1]"/>
      <arg name="sensor_camera_scene_topics" default="[airsim/leftcamera/rgb/image, airsim/rightcamera/rgb/image]"/>
      <arg name="sensor_camera_segmentation_topics" default="[airsim/leftcamera/segmentation/image, airsim/rightcamera/segmentation/image]"/>
      <arg name="sensor_camera_depth_topics" default="[airsim/leftcamera/depth/image, airsim/rightcamera/depth/image]"/>
      <arg name="sensor_camera_frames" default="[base_leftcamera, base_rightcamera]"/>

      <arg name="object_names" default="[object_name]"/>
      <arg name="objects_coordinates_local" default="[1]"/>
      <arg name="object_topics" default="[airsim/object/object_name/pose]"/>

      <node name="airsim_publish" pkg="airsim" type="airsim_publish.py" respawn="false" output="screen">

            <param name="rate" type="int" value="10"/>

            <param name="ip" type="string" value="" />

            <param name="toggle_drone" type="int" value="0" />

            <param name="vehicle_name" type="string" value="airsimvehicle" />
            <param name="vehicle_base_frame" type="string" value="base_link" />

            <param name="pose_topic" type="string" value="airsim/gtpose" />
            <param name="pose_frame" type="string" value="world" />

            <param name="sensor_imu_enable" type="int" value="1" />
            <param name="sensor_imu_name" type="string" value="imu" />
            <param name="sensor_imu_topic" type="string" value="airsim/imu" />
            <param name="sensor_imu_frame" type="string" value="world" />

            <rosparam param="sensor_echo_names" subst_value="True">$(arg sensor_echo_names)</rosparam>
            <rosparam param="sensor_echo_topics" subst_value="True">$(arg sensor_echo_topics)</rosparam>
            <rosparam param="sensor_echo_frames" subst_value="True">$(arg sensor_echo_frames)</rosparam>

            <rosparam param="sensor_lidar_names" subst_value="True">$(arg sensor_lidar_names)</rosparam>
            <rosparam param="sensor_lidar_toggle_groundtruth" subst_value="True">$(arg sensor_lidar_toggle_groundtruth)</rosparam>
            <rosparam param="sensor_lidar_topics" subst_value="True">$(arg sensor_lidar_topics)</rosparam>
            <rosparam param="sensor_lidar_segmentation_topics" subst_value="True">$(arg sensor_lidar_segmentation_topics)</rosparam>
            <rosparam param="sensor_lidar_frames" subst_value="True">$(arg sensor_lidar_frames)</rosparam>

            <rosparam param="sensor_gpulidar_names" subst_value="True">$(arg sensor_gpulidar_names)</rosparam>
            <rosparam param="sensor_gpulidar_topics" subst_value="True">$(arg sensor_gpulidar_topics)</rosparam>
            <rosparam param="sensor_gpulidar_frames" subst_value="True">$(arg sensor_gpulidar_frames)</rosparam>

            <rosparam param="sensor_uwb_names" subst_value="True">$(arg sensor_uwb_names)</rosparam>
            <rosparam param="sensor_uwb_topics" subst_value="True">$(arg sensor_uwb_topics)</rosparam>
            <rosparam param="sensor_uwb_frames" subst_value="True">$(arg sensor_uwb_frames)</rosparam>

            <rosparam param="sensor_camera_names" subst_value="True">$(arg sensor_camera_names)</rosparam>
            <rosparam param="sensor_camera_toggle_scene_mono" subst_value="True">$(arg sensor_camera_toggle_scene_mono)</rosparam>
            <rosparam param="sensor_camera_scene_quality" subst_value="True">$(arg sensor_camera_scene_quality)</rosparam>
            <rosparam param="sensor_camera_toggle_segmentation" subst_value="True">$(arg sensor_camera_toggle_segmentation)</rosparam>
            <rosparam param="sensor_camera_toggle_depth" subst_value="True">$(arg sensor_camera_toggle_depth)</rosparam>
            <rosparam param="sensor_camera_scene_topics" subst_value="True">$(arg sensor_camera_scene_topics)</rosparam>
            <rosparam param="sensor_camera_segmentation_topics" subst_value="True">$(arg sensor_camera_segmentation_topics)</rosparam>
            <rosparam param="sensor_camera_depth_topics" subst_value="True">$(arg sensor_camera_depth_topics)</rosparam>
            <rosparam param="sensor_camera_frames" subst_value="True">$(arg sensor_camera_frames)</rosparam>

            <rosparam param="object_names" subst_value="True">$(arg object_names)</rosparam>
            <rosparam param="objects_coordinates_local" subst_value="True">$(arg objects_coordinates_local)</rosparam>
            <rosparam param="object_topics" subst_value="True">$(arg object_topics)</rosparam>
      </node>
</launch>
